blockmap.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+const BlockMap = require('blockmap');
+const crypto = require('crypto');
+
+function getNotDiscardedChunks(disk, blockSize, capacity) {
+	const chunks = [];
+	const discardedChunks = disk.getDiscardedChunks();
+	let lastStart = 0;
+	for (let discardedChunk of discardedChunks) {
+		chunks.push([lastStart, discardedChunk.start - 1]);
+		lastStart = discardedChunk.end + 1;
+	}
+	if (lastStart < capacity) {
+		chunks.push([lastStart, capacity - 1]);
+	}
+	return chunks;
+}
+
+function mergeBlocks(blocks) {
+	if (blocks.length > 1) {
+		let last = blocks[0];
+		for (let i = 1; i < blocks.length; i++) {
+			let block = blocks[i];
+			if ((block[0] >= last[0]) && (block[0] <= last[1] + 1)) {
+				last[1] = block[1];
+				blocks.splice(i, 1);
+				i--;
+			} else {
+				last = block;
+			}
+		}
+	}
+}
+
+function streamSha256(stream, callback) {
+	const hash = crypto.createHash('sha256');
+	stream.on('data', function(data) {
+		hash.update(data);
+	});
```

Comment:
@jhermsmeier 
this doesn't make the code shorter or more readable, but we can use that if you prefer: 

```javascript
function streamSha256(stream, callback) {
	const hash = crypto.createHash('sha256');
	hash.setEncoding('hex');
	stream.pipe(hash);
	stream.on('error', callback);
	stream.on('end', function() {
		hash.end()
		callback(null, hash.read());
	});
}
```